log/field.go

```diff
@@ -0,0 +1,204 @@
+package log
```

Comment:
It's true... I dunno, seems like there are reasonable arguments in either direction. I would have liked to add a comment about supporting `Reference` types in the N-1 other languages (which have been submitted already). How strongly do you feel about this? (I'm on the fence)